examples/platforms/simulation/virtual_time/platform-sim.c

```diff
@@ -288,7 +298,8 @@ void otSysProcessDrivers(otInstance *aInstance)
     platformUartUpdateFdSet(&read_fds, &write_fds, &error_fds, &max_fd);
 #endif
 
-    if (!otTaskletsArePending(aInstance) && platformAlarmGetNext() > 0 && !platformRadioIsTransmitPending())
+    if (!otTaskletsArePending(aInstance) && platformAlarmGetNext() > 0 &&
+        (!platformRadioIsTransmitPending() || platformRadioTaskPending()))
```

Comment:
According to doxygen of `platformRadioTaskPending`, it's also waiting for radio TX done, which looks like same as `platformRadioIsTransmitPending`?

Reply:
It seems similar. Maybe the description was not very clear, but the fact is that:
- ```platformRadioIsTransmitPending``` deals with process that are internal to the device, making it transmit when desired;
- ```platformRadioTaskPending``` deals with process related to the signal exchange between the device and the simulator.

Thanks for observing this.